modules/tinymce/src/plugins/accordion/main/ts/core/Actions.ts

```diff
@@ -0,0 +1,29 @@
+import Editor from 'tinymce/core/api/Editor';
+
+const insertAccordion = (editor: Editor): void => {
+  const dom = editor.dom;
+
+  const rng = editor.selection.getRng();
+  const summaryText = rng.collapsed ? 'Accordion summary...' : rng.toString();
+
+  const target = dom.getParent(rng.commonAncestorContainer, dom.isBlock);
```

Comment:
This is to wide a block can be things like tables and lists so this breaks the structure if you start wrapping parts of the table or list structure in details elements. It think this should either be a smaller set of block types, the schema has a text blocks list, or it would have to be smarter on the types of blocks that do break. Also in TDs the elements might not have any blocks so it would have to be smart to first take the contents in the cell and convert that into a details body elements.